src/main/java/org/opentripplanner/routing/graph/Graph.java

```diff
@@ -682,6 +682,11 @@ public static Graph load(File file) throws IOException {
      * TODO: do we really need a factory for different street vertex indexes?
      */
     public void index (StreetVertexIndexFactory indexFactory) {
+        if (streetIndex != null) {
+            throw new UnsupportedOperationException(
+                "A streetIndex has already been defined. Can't you just reuse the existing one?"
```

Comment:
I added a snarky error message here, as I don't think there would be a legitimate reason to re-index a graph, but maybe I'm wrong.

Reply:
Is this one of the checks to ensure that constraints are respected (that there's only one spatial index created per graph)? If so it would be good to explain to the person seeing the message why they should just reuse the existing one, and why it's unwise to create another one.

Reply:
Yes, this is a check that was added to prevent the creation of more StreetIndexes in downstream code. However, I honestly don't know if there are benefits to reindexing a graph, so I could use some guidance here.